### PR TITLE
Gas Optimization Bounty

### PR DESCRIPTION
**Pull request by malik672**
Optimized the contracts

### CODE REVIEW

**Comment by nkrishang** [comment]
Hey @malik672 👋🏽 I want to get your optimizations merged.

I see 122 files changed. Could you summarize the change(s) you've made to the contracts, something like this bounty PR https://github.com/thirdweb-dev/contracts/pull/463 which recently got merged? We'll put out a PR template this week so this part is clear for future bounty submissions.

Additionally, could you please resolve merge conflicts? I can then get to this PR asap. Thanks!

**Comment by malik672** [comment]
ok
